sentry_streams/sentry_streams/k8s/generate.py

```diff
@@ -0,0 +1,197 @@
+import argparse
+import copy
+import importlib.resources
+import sys
+from typing import Any, List, Mapping
+
+import yaml
+
+from sentry_streams.adapters.stream_adapter import PipelineConfig
+
+K8sDeploymentManifest = Mapping[str, Any]
+K8sConfigMapManifest = Mapping[str, Any]
+
+
+def _merge_labels(
+    tpl_labels: Mapping[str, str], seg_labels: Mapping[str, str]
+) -> Mapping[str, str]:
+    """
+    Merge labels. Segment labels overrides the template labels.
+    """
+    return {**tpl_labels, **seg_labels}
+
+
+def apply_namespace(
+    k8s_resources: List[K8sConfigMapManifest | K8sDeploymentManifest],
+    namespace: str | None = None,
+) -> None:
+    for k8s_resource in k8s_resources:
+        k8s_resource["metadata"]["namespace"] = namespace
+
+
+def generate_configmap(
+    *, config: PipelineConfig, configmap_template: K8sConfigMapManifest
+) -> K8sConfigMapManifest:
+    configmap = copy.deepcopy(configmap_template)
+    pipeline_name = config["pipeline"]["name"]
+    configmap["metadata"]["name"] = pipeline_name
```

Comment:
I think we may want to call the configmap `pipeline_<PIPELINE_NAME>` rather than `PIPELINE_NAME` as it has to be a unique name in the namespace.